server/models/Transaction.ts

```diff
@@ -1459,7 +1459,7 @@ Transaction.createFromContributionPayload = async (
 };
 
 Transaction.createActivity = (transaction, options) => {
-  if (transaction.deletedAt) {
+  if (transaction.deletedAt || ['TAX', 'PAYMENT_PROCESSOR_FEE'].includes(transaction.kind)) {
```

Comment:
You could use constants instead of strings here.

Reply:
Now that the model is typed with Typescript, I don't see using constants as a strong requirement; it makes the code more verbose, and we'll get an error if there's a typo in the string anyway.